src/components/send/components/AccountIDInput.js

```diff
@@ -0,0 +1,214 @@
+        const { handleChange, localAlert, clearLocalAlert } = this.props
+        const { wrongChar } = this.state
+        const pattern = /[^a-zA-Z0-9._-]/;
+
+        const accountId = userValue.trim().toLowerCase()
+
+        if (accountId.match(pattern)) {
+            if (wrongChar) {
+                el.style.animation = 'none'
+                void el.offsetHeight
+                el.style.animation = null
+            } else {
+                this.setState({ wrongChar: true })
+            }
+            return;
+        } else {
+            this.setState({ wrongChar: false })
+        }
+
+        handleChange(accountId)
+
+        localAlert && clearLocalAlert()
+
+        this.checkAccountAvailabilityTimer && clearTimeout(this.checkAccountAvailabilityTimer)
+        this.checkAccountAvailabilityTimer = setTimeout(() => {
+            this.handleCheckAvailability(accountId)
+        }, ACCOUNT_CHECK_TIMEOUT)
+    }
+
+    isImplicitAccount = (accountId) => accountId.length === 64 && !accountId.includes('.')
+
+    handleCheckAvailability = async (accountId) => {
+        const { checkAvailability, clearLocalAlert } = this.props
+
+        if (!accountId) {
+            return false
+        }
+
+        try {
+            await checkAvailability(accountId)
+        } catch(e) {
+            if (this.isImplicitAccount(accountId) && e.toString().includes('does not exist while viewing')) {
+                console.warn(`${accountId} does not exist. Assuming this is an implicit Account ID.`)
+                clearLocalAlert()
+                return;
+            }
+            throw e;
+        }
+    }
+
+    render() {
+        const {
+            disabled,
+            localAlert,
+            accountId
+        } = this.props
+
+        const { wrongChar } = this.state
+        const success = localAlert?.success
+        const problem = !localAlert?.success && localAlert?.show
+
+        return (
+            <Translate>
+                {({ translate }) => (
+                    <InputWrapper className={classNames([{ 'success': success }, { 'problem': problem }, { 'wrong-char': wrongChar }])}>
+                        <input
+                            ref={this.input}
+                            value={accountId}
+                            onInput={(e) => this.updatePrefix(e.target.value)}
+                            onChange={e => this.handleChangeAccountId({ userValue: e.target.value, el: e.target })}
+                            placeholder={translate('input.accountId.placeholder')}
+                            required
+                            autoComplete='off'
+                            autoCorrect='off'
+                            autoCapitalize='off'
+                            spellCheck='false'
+                            tabIndex='1'
+                            disabled={disabled}
+                        />
+                        <span className='success-prefix' ref={this.prefix}>
```

Comment:
Not a blocker, but this also could be prop driven rather than using a `ref`, since all we are doing with it is setting some styles on it based on props/state.